src/qoreprocesshandler.h

```diff
@@ -1,97 +1,85 @@
+                       ResolvedCallReferenceNode* on_setup,
+                       ResolvedCallReferenceNode* on_error,
+                       ResolvedCallReferenceNode* on_fork_error,
+                       ResolvedCallReferenceNode* on_exec_setup,
+                       ResolvedCallReferenceNode* on_exec_error) :
+        m_xsink(xsink),
+        m_on_success(on_success, xsink),
+        m_on_setup(on_setup, xsink),
+        m_on_error(on_error, xsink),
+        m_on_fork_error(on_fork_error, xsink),
+        m_on_exec_setup(on_exec_setup, xsink),
+        m_on_exec_error(on_exec_error, xsink)
     {
     }
 
     template<typename Executor>
-    void on_success(Executor & exec) const {
-        call("on_success", m_on_success, exec, std::error_code());
+    void on_success(Executor& exec) const {
+        call("on_success", *m_on_success, exec, std::error_code());
     }
 
     template<typename Executor>
-    void on_setup(Executor & exec) const {
-        call("on_setup", m_on_setup, exec, std::error_code());
+    void on_setup(Executor& exec) const {
+        call("on_setup", *m_on_setup, exec, std::error_code());
     }
 
     template<typename Executor>
-    void on_error(Executor & exec, const std::error_code &ec) const {
-        call("on_error", m_on_error, exec, ec);
+    void on_error(Executor& exec, const std::error_code& ec) const {
+        call("on_error", *m_on_error, exec, ec);
     }
 
     template<typename Executor>
-    void on_fork_error(Executor & exec, const std::error_code & ec) const {
-        call("on_fork_error", m_on_fork_error, exec, ec);
+    void on_fork_error(Executor& exec, const std::error_code& ec) const {
+        call("on_fork_error", *m_on_fork_error, exec, ec);
     }
 
     template<typename Executor>
-    void on_exec_setup(Executor & exec) const {
-        call("on_exec_setup", m_on_exec_setup, exec, std::error_code());
+    void on_exec_setup(Executor& exec) const {
+        call("on_exec_setup", *m_on_exec_setup, exec, std::error_code());
     }
 
     template<typename Executor>
-    void on_exec_error(Executor & exec, const std::error_code &ec) const {
-        call("on_exec_error", m_on_exec_error, exec, ec);
+    void on_exec_error(Executor& exec, const std::error_code& ec) const {
+        call("on_exec_error", *m_on_exec_error, exec, ec);
     }
 
     template<typename Executor>
     void call(const char* info,
-              const ResolvedCallReferenceNode *callref,
-              Executor & exec,
-              const std::error_code &ec
-              ) const
+              const ResolvedCallReferenceNode* callref,
+              Executor& exec,
+              const std::error_code& ec) const
     {
         if (!callref) {
             //std::cout << "    info: " << info << ": no handler installed" << std::endl;
             return;
         }
 
-        QoreHashNode *e = new QoreHashNode(autoTypeInfo);
+        QoreHashNode* e = new QoreHashNode(autoTypeInfo);
```

Comment:
what if any of the next lines throw an exception, will QoreHashNode be freed? Shouldn't we use a reference holder or smth like that, before we pass it into the args?
Also, please give it a better name than `e` :)